lightning_client.go

```diff
@@ -2733,6 +2741,37 @@ func (s *lightningClient) OpenChannel(ctx context.Context, peer route.Vertex,
 	}, nil
 }
 
+// OpenChannelStream opens a channel to the specified peer and with the
+// specified arguments and options. This function returns a stream of
+// updates.
+func (s *lightningClient) OpenChannelStream(ctx context.Context, peer route.Vertex,
+	localSat, pushSat btcutil.Amount, private bool,
+	opts ...OpenChannelOption) (lnrpc.Lightning_OpenChannelClient, error) {
+
+	rpcCtx, cancel := context.WithTimeout(ctx, s.timeout)
+	defer cancel()
+
+	rpcRequest := &lnrpc.OpenChannelRequest{
+		NodePubkey:         peer[:],
+		LocalFundingAmount: int64(localSat),
+		PushSat:            int64(pushSat),
+		Private:            private,
+	}
+
+	for _, opt := range opts {
+		opt(rpcRequest)
+	}
+
+	rpcCtx = s.adminMac.WithMacaroonAuth(rpcCtx)
+
+	stream, err := s.client.OpenChannel(rpcCtx, rpcRequest)
```

Comment:
I guess the "main purpose" of `lndclient` is to abstract away some of the gRPC stream handling logic. So it would make sense to return native Golang channels here instead of the raw stream. See `SubscribeChannelEvents` for an example.